pandas/core/dtypes/dtypes.py

```diff
@@ -358,11 +358,11 @@ def construct_from_string(cls, string):
         try:
             if string == 'category':
                 return cls()
-        except:
+            else:
```

Comment:
I think this *may* have been it. For whatever reason, we're here with

```
ipdb> pp string
   A  A
0  0  2
1  2  4
2  0  4
```

We do

```
ipdb> string == 'category'
*** FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
```

That FutureWarning is elevated to an error. I didn't realize this, but you can catch warnings, so the very simple fix is to also catch `FutureWarning` here, but that feels fragile. I'll investigate a better fix...